akka-stream-tests/src/test/scala/akka/stream/impl/fusing/AsyncCallbackSpec.scala

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.impl.fusing
+
+import akka.Done
+import akka.actor.ActorRef
+import akka.stream.stage._
+import akka.stream._
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.stream.testkit.{ TestPublisher, TestSubscriber }
+import akka.testkit.{ AkkaSpec, TestProbe }
+
+import scala.concurrent.{ Future, Promise }
+
+class AsyncCallbackSpec extends AkkaSpec {
+
+  implicit val materializer = ActorMaterializer(ActorMaterializerSettings(system).withFuzzing(false))
+
+  case object Started
+  case class Elem(n: Int)
+  case object Stopped
+
+  class AsyncCallbackGraphStage(probe: ActorRef, early: Option[AsyncCallback[AnyRef] ⇒ Unit] = None)
+    extends GraphStageWithMaterializedValue[FlowShape[Int, Int], Future[AsyncCallback[AnyRef]]] {
```

Comment:
why `Future`, wasn't one of the main features of the new callbacks that they can be used immediately?

Reply:
Hmm, yeah, I think you are right, I probably did it like this out of habit. I'll try to skip the future and see.

Reply:
Hehe yeah I actually noticed this too, but thought it's fine; also known as the "glorified var masked in a Promise pattern" ;-) Just a var will do fine here...

Reply:
Actually, no var needed even.